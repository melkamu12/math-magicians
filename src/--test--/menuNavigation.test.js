import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MainNavigation from '../components/mainNavigation';
import '@testing-library/jest-dom/extend-expect';

describe('Navigation menu component', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <MainNavigation />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Navigation menu component correctly', () => {
    render(
      <MemoryRouter>
        <MainNavigation />
      </MemoryRouter>,
    );
    const homeLink = screen.getByText(/Home\b/i);
    const calculatorLink = screen.getByText(/Calculator\b/i);
    const quoteLink = screen.getByText(/Quote\b/i);

    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute('href', '/');
    expect(calculatorLink).toBeInTheDocument();
    expect(calculatorLink).toHaveAttribute('href', '/MathCalculator');
    expect(quoteLink).toBeInTheDocument();
    expect(quoteLink).toHaveAttribute('href', '/Quote');
  });
});
