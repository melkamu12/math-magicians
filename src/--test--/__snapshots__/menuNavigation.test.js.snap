// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation menu component should match DOM Snapshot 1`] = `
<header
  className="header"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul
      className="List"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home |
        </a>
      </li>
      <li>
        <a
          href="/MathCalculator"
          onClick={[Function]}
        >
          Calculator |
        </a>
      </li>
      <li>
        <a
          href="/Quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
